frontend/packages/dev-console/src/components/edit-deployment/utils/edit-deployment-utils.ts

```diff
@@ -8,3 +29,501 @@ export const getContainerNames = (containers: { [key: string]: any }[]) => {
     }, {}) ?? []
   );
 };
+
+export const getLchImageStreamData = (
+  resName: string,
+  resNamespace: string,
+  tagImages?: { containerName: string; to: { [key: string]: string } }[],
+): LifecycleHookImagestreamData => {
+  const imageTagNs = tagImages?.[0]?.to?.namespace ?? '';
+  const image = tagImages?.[0]?.to?.name?.split(':') ?? [];
+  return {
+    name: resName,
+    project: { name: resNamespace },
+    fromImageStreamTag: true,
+    imageStreamTag: {},
+    containerName: tagImages?.[0]?.containerName ?? '',
+    to: tagImages?.[0]?.to ?? {},
+    isSearchingForImage: false,
+    imageStream: {
+      namespace: imageTagNs || resNamespace,
+      image: image[0] ?? '',
+      tag: image[1] ?? '',
+    },
+    isi: {
+      name: '',
+      image: {},
+      tag: '',
+      status: { metadata: {}, status: '' },
+      ports: [],
+    },
+    image: {
+      name: '',
+      image: {},
+      tag: '',
+      status: { metadata: {}, status: '' },
+      ports: [],
+    },
+  };
+};
+
+export const getLifecycleHookData = (lch: any): LifecycleHookData => {
+  return {
+    failurePolicy: lch?.failurePolicy ?? FailurePolicyType.Abort,
+    execNewPod: {
+      command: lch?.execNewPod?.command ?? [''],
+      containerName: lch?.execNewPod?.containerName,
+      env: lch?.execNewPod?.env,
+      volumes: _.join(lch?.execNewPod?.volumes, ','),
+    },
+    tagImages: lch?.tagImages ?? [],
+  };
+};
+
+export const getLifecycleHookFormData = (lch: any): LifecycleHookFormData => {
+  return {
+    lch: getLifecycleHookData(lch),
+    exists: !!lch,
+    isAddingLch: false,
+    action: lch
+      ? lch.hasOwnProperty(LifecycleAction.execNewPod)
+        ? LifecycleAction.execNewPod
+        : LifecycleAction.tagImages
+      : LifecycleAction.execNewPod,
+  };
+};
+
+export const getStrategyData = (
+  type: string,
+  strategy: any,
+  resName: string,
+  resNamespace: string,
+  resourceType: string,
+) => {
```

Comment:
Is it possible to get a return type for this? Or will that be messy? It'll make it easier to call upon this method in the future if the full signature is noted.